Polygon_repair/doc/Polygon_repair/Polygon_repair.txt

```diff
@@ -63,77 +68,63 @@ Invalid: (a) self-intersecting polygon self-intersection, (b) self-touching poly
 
-For the purposes of the repair operation, the input polygon, polygon with holes
-or multipolygon is merely used as a container of input line segments. These line
-segments are added to the arrangement as edges. Internally, this is done using
-a constrained triangulation where they are added as constraints.
+\section SectionPolygonRepair_UnionIntersection Union and Intersection Rule
 
-With the even-odd rule, only the edges that are present an odd number of
-times in the input will be edges in the final arrangement.
-When these edges are only partially overlapping, only the parts that overlap
-an odd number of times will be edges in the final arrangement.
+Given several valid polygons this rule computes their union or intersection.
+This approximates from outside and from inside, respectively.
 
-This procedure is done in two steps: 1. preprocessing to eliminate identical
-edges that are present an even number of times, and 2. adding edges incrementally
-while applying an even-odd counting mechanism, which erases existing (parts of)
-edges when new overlapping ones are added.
+\cgalFigureBegin{UnionIntersection, UnionIntersection.svg}
+Union (top) and Intersection (bottom).
+\cgalFigureEnd
 
-\subsection SubsectionPolygonRepair_Labeling Labeling
 
-First, the polygon exterior is labeled. For this, all of the faces that can be
-accessed from the exterior without passing through an edge are labeled as exterior
-faces.
 
-Then, all other faces are labeled. For the even-odd rule, the label applied
-alternates between polygon interior and hole every time that an edge is passed.
+\section SubsectionPolygonRepair_Output Notes on the Output
 
-\subsection SubsectionPolygonRepair_Reconstruction Reconstruction of the Multipolygon
+The conditions listed above are sufficient to define valid polygons, polygons
+with holes and multipolygons with holes for most applications. However, in
+order to ensure unique deterministic output from the repair algorithm,
+the valid multipolygons with holes returned by the package conform to more
+strict criteria:
 
-The algorithm reconstructs the multipolygon boundary by boundary, obtaining
-counter-clockwise cycles for outer boundaries and clockwise cycles for inner
-boundaries. Once all boundaries have been reconstructed, the boundaries are assembled
-into multipolygons using the face labels to know which polygon with holes inner/outer
-boundaries belong to, and using the orientation to distinguish between the outer and
-inner boundaries of each polygon with holes.
+- Adjacent collinear edges touching at vertices of degree two are merged
+- The sequence of vertices representing a boundary starts from its
+lexicographically smallest vertex
+- Outer boundaries are oriented counter-clockwise and inner boundaries are
+oriented clockwise
+- The inner boundaries of a polygon with holes are stored in lexicographic
+order
+- The polygons with holes of a multipolygon with holes are also stored in
+lexicographic order
```

Comment:
line 131, it is now strange to reference the section we are already in. Remove `\ref SubsectionPolygonRepair_Output`